server/src/main/java/org/opensearch/search/query/QueryCollectorContext.java

```diff
@@ -248,7 +249,9 @@ CollectorManager<? extends Collector, ReduceableSearchResult> createManager(
                 CollectorManager<? extends Collector, ReduceableSearchResult> in
             ) throws IOException {
                 final List<CollectorManager<?, ReduceableSearchResult>> managers = new ArrayList<>();
-                managers.add(in);
+                if (Objects.nonNull(in)) {
```

Comment:
Under which conditions `in` is `null`?

Reply:
As per logic in [QueryCollectorManagerContext](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/query/QueryCollectorManagerContext.java#L71) this is `null` when we iterate over collection of collectorContexts and this is first collectContext in that collection.
In my test that happened if I'm skip adding the TopDocsCollector

Reply:
I've removed this check